src/features/autofraction.ts

```diff
@@ -94,17 +94,22 @@ export const runAutoFractionCursor = (view: EditorView, ctx: Context, range: Sel
 		}
 	}
 
-	// Run autofraction
-	let numerator = view.state.sliceDoc(start, to);
-
 	// Don't run on an empty line
-	if (numerator === "") return false;
-
+	if (start === to) { return false; }
+
+	// Remove unnecessary outer parentheses
+	const { beg, end } = (() => {
+		if (curLine.charAt(start) === "(" && curLine.charAt(to - 1) === ")") {
+			const closing = findMatchingBracket(curLine, start, "(", ")", false, to);
+			if (closing === to - 1) {
```

Comment:
I think this would be cleaner if we kept the original code, and simply moved the `findMatchingBracket` check inside the initial `if (curLine.charAt(start) === "(" && curLine.charAt(to - 1) === ")")` check?

Reply:
at the moment there's an implicit dependency between `numerator`, `curLine`, `start`, `to`; as i have it written at the moment mutation is minimized (no `let`s) and less string copying involved, albeit at the expense of being somewhat ugly with the iife. but even if that's removed i think i'd still favor something like
```js
if (start === to) { return false; }

let numerator = view.state.sliceDoc(beg, end);
if (numerator.at(0) === "(" && numerator.at(-1) === ")") {
    numerator = numerator.slice(1, -1);
}
```
to reduce some of those implicit dependencies - namely, still using `start === to` as the empty check and using `numerator.at()` to check its wrapping parentheses, as opposed to using `curLine`, `start`, and `to`, and i think this code still fairly resembles the original.

Reply:
Ah, yes; using `start === to` and `numerator.at()` is definitely an improvement. I'd prefer to let `numerator` be mutable though; imo it's cleaner and involves less nested code.